### PR TITLE
Registry option to disable mouse capture

### PR DESCRIPTION
**Pull request by michalpelka**
## What does this PR do?

This PR allows setting a registry key "O3DE/InputSystem/Mouse/CaptureMouseCursor" to disable mouse capture.
It is useful in two scenarios:
 - during a debugging session, where you can be left with the system without a cursor and active breakpoint
 - simulation use case, where switching context and focus does not always work, and the user is left with the GameLauncher window minimized and without a mouse cursor.

It is supported on Linux and Windows. I have no capability to implement MacOS version.
## How was this PR tested?

- [x] Build and test against Linux GameLauncher
- [x] Build and test against Windows GameLauncher

### CODE REVIEW

**Comment by michalpelka** [comment]
I've tested it on Windows on the empty project, and works as expected. I kindly ask for review of this feature.

**Review by spham-amzn — APPROVED** [review]

File: Code/Framework/AzFramework/AzFramework/Input/Devices/Mouse/InputDeviceMouse.h [hunk 1]
```diff
@@ -189,6 +189,11 @@ namespace AzFramework
         //! \param[in] sampleRateHertz The raw movement sample rate in Hertz (cycles per second)
         void SetRawMovementSampleRate(AZ::u32 sampleRateHertz);
 
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! Set if the cursor should be visible or hidden
+        //! \param[in] visible True if the cursor should be visible, false if it should be hidden
```
Comment (pijaro) [inline Code/Framework/AzFramework/AzFramework/Input/Devices/Mouse/InputDeviceMouse.h:194]:
```suggestion
        //! \param[in] captureCursor True if the cursor should be visible, false if it should be hidden
```

File: Code/Framework/AzFramework/AzFramework/Input/Devices/Mouse/InputDeviceMouse.h [hunk 2]
```diff
@@ -273,6 +278,11 @@ namespace AzFramework
             //! \param[in] sampleRateHertz The raw movement sample rate in Hertz (cycles per second)
             void SetRawMovementSampleRate(AZ::u32 sampleRateHertz);
 
+            ////////////////////////////////////////////////////////////////////////////////////////////
+            //! Set if the cursor should be visible or hidden
+            //! \param[in] visible True if the cursor should be visible, false if it should be hidden
```
Comment (pijaro) [inline Code/Framework/AzFramework/AzFramework/Input/Devices/Mouse/InputDeviceMouse.h:283]:
```suggestion
            //! \param[in] captureCursor True if the cursor should be visible, false if it should be hidden
```

File: Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceMouse.cpp [hunk 3]
```diff
@@ -13,6 +13,7 @@
 
 namespace AzFramework
 {
+
```
Comment (pijaro) [inline Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceMouse.cpp:16]: Nit: Is this newline necessary?

**Review by pijaro — COMMENTED** [review]
This is a good feature, nice job. Just few nitpicks.

**Review by adamdbrw — APPROVED** [review]

File: Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceMouse.cpp (hunk 3 above)
Comment (michalpelka) [inline Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbInputDeviceMouse.cpp:16]:
```suggestion
```

**Pull request merged**